.github/actions/build/action.yml

```diff
@@ -42,14 +45,13 @@ runs:
 
     - name: Docker build
       id: build
-      uses: docker/build-push-action@v2
+      uses: docker/build-push-action@v4
       with:
         context: ${{ inputs.context }}
         tags: ${{ inputs.tags }}
         build-args: ${{ inputs.build_args }}
-        # @see - https://docs.docker.com/build/ci/github-actions/cache/#github-cache
-        cache-from: type=gha
-        cache-to: type=gha,mode=max
+        cache-from: type=gha,scope=$GITHUB_REF_NAME-${{ inputs.app_name }}
+        cache-to: type=gha, mode=max,scope=$GITHUB_REF_NAME-${{ inputs.app_name }}
```

Comment:
Remove space from within - `type=gha, mode=max`

Reply:
`$GITHUB_REF_NAME` might not work here, should use `${{ github.xxx (look around see what we should use) }}` instead.